runtime/doc/msgpack_rpc.txt

```diff
@@ -45,19 +45,19 @@ non-static API functions, converting/validating arguments and return values
 back to msgpack.
 
 Client libraries will normally provide wrappers that hide msgpack-rpc details
-from programmers, which can be automatically generated by reading bundled API
-metadata from a compiled nvim instance.
+from programmers. The wrappers can be automatically generated by reading
+bundled API metadata from a compiled Nvim instance.
 
 There are two ways to obtain API metadata:
 
-1. By connecting to a running nvim instance and calling `vim_get_api_info`
-   via msgpack-rpc. This is the preferred way for clients written in
-   dynamically-typed languages, which can define functions at runtime.
+1. By connecting to a running Nvim instance and calling `vim_get_api_info`
+   via msgpack-rpc. This is best for clients written in dynamically-typed
+   languages, which can define functions at runtime.
 
-2. Through the `--api-info` command-line option, which makes nvim dump a
-   msgpack blob containing metadata to stdout and exit. This is preferred
-   when writing clients for statically-typed languages, which require a
-   separate compilation step.
+2. By starting Nvim with the `--api-info` command-line option, which makes Nvim
+   dump a msgpack metadata to standard output and exit. This is best for clients
+   written in statically-typed languages, which require a separate compilation
```

Comment:
Should this be "which *often* require a separate compilation step"?

I'm not actually sure about why a separate compilation step matters. Should it say that functions can't be defined after compile time?